core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java

```diff
@@ -2987,8 +2987,12 @@ public static void setLocalDictColumnsToWrapperSchema(List<ColumnSchema> columns
     if (null != localDictExcludeColumns) {
       listOfDictionaryExcludeColumns = localDictExcludeColumns.trim().split("\\s*,\\s*");
     }
-    if (null != isLocalDictEnabledForMainTable && Boolean
-        .parseBoolean(isLocalDictEnabledForMainTable)) {
+    if (null == isLocalDictEnabledForMainTable) {
```

Comment:
This change is not required if create table command is taking care of setting default value for  isLocalDictEnabledForMainTable